src/box/lua/space.cc

```diff
@@ -765,12 +765,16 @@ box_lua_space_delete(struct lua_State *L, uint32_t id)
 static int
 box_lua_space_new_or_delete(struct trigger *trigger, void *event)
 {
-	struct lua_State *L = (struct lua_State *) trigger->data;
+	(void)trigger;
+	struct lua_State *L = tarantool_L;
 	struct space *space = (struct space *) event;
 
 	if (space_by_id(space->def->id) != NULL) {
 		box_lua_space_new(L, space);
 	} else {
+		/* tarantool_L is freed on Tarantool shutdown. */
+		if (L == NULL)
+			return 0;
```

Comment:
Shouldn't box be destroyed before Lua? I don't like that there are so many places now where we have to check `tarantool_L`.